curvefs/src/metaserver/inode_storage.cpp

```diff
@@ -194,38 +197,96 @@ MetaStatusCode InodeStorage::AddS3ChunkInfoList(
                     MetaStatusCode::STORAGE_INTERNAL_ERROR;
 }
 
-// NOTE: s3chunkinfo which its chunkid equal or
-// less then min chunkid should be removed
-MetaStatusCode InodeStorage::RemoveS3ChunkInfoList(Transaction txn,
-                                                   uint32_t fsId,
-                                                   uint64_t inodeId,
-                                                   uint64_t chunkIndex,
-                                                   uint64_t minChunkId,
-                                                   uint64_t* size4del) {
+bool InodeStorage::SplitS3ChunkInfoList(const std::string& value,
+                                        uint64_t delFirstChunkId,
+                                        S3ChunkInfoList* list2add) {
+    if (!conv_->ParseFromString(value, list2add)) {
+        LOG(ERROR) << "ParseFromString failed for S3ChunkInfoList";
+        return false;
+    }
+
+    size_t size = list2add->s3chunks_size();
+    for (size_t i = 0; i < size; i++) {
+        if (list2add->s3chunks(i).chunkid() >= delFirstChunkId) {
+            list2add->mutable_s3chunks()->DeleteSubrange(i, size - i);
+            break;
+        }
+    }
+    return true;
+}
+
+MetaStatusCode InodeStorage::DelS3ChunkInfoList(
+    Transaction txn,
+    uint32_t fsId,
+    uint64_t inodeId,
+    uint64_t chunkIndex,
+    const S3ChunkInfoList& list2del) {
+    if (list2del.s3chunks_size() == 0) {
+        return MetaStatusCode::OK;
+    }
+
+    size_t size = list2del.s3chunks_size();
+    uint64_t delFirstChunkId = list2del.s3chunks(0).chunkid();
+    uint64_t delLastChunkId = list2del.s3chunks(size - 1).chunkid();
+
+    // prefix
     Prefix4ChunkIndexS3ChunkInfoList prefix(fsId, inodeId, chunkIndex);
     std::string sprefix = conv_->SerializeToString(prefix);
     auto iterator = txn->SSeek(table4s3chunkinfo_, sprefix);
     if (iterator->Status() != 0) {
         return MetaStatusCode::STORAGE_INTERNAL_ERROR;
     }
 
-    *size4del = 0;
-    uint64_t lastChunkId;
     Key4S3ChunkInfoList key;
     std::vector<std::string> key2del;
+    std::vector<S3ChunkInfoList> lists2add;
     for (iterator->SeekToFirst(); iterator->Valid(); iterator->Next()) {
         std::string skey = iterator->Key();
         if (!StringStartWith(skey, sprefix)) {
             break;
         } else if (!conv_->ParseFromString(skey, &key)) {
             return MetaStatusCode::PARSE_FROM_STRING_FAILED;
-        } else if (key.firstChunkId >= minChunkId) {
-            break;
         }
 
-        // firstChunkId < minChunkId
-        key2del.push_back(skey);
-        *size4del += key.size;
+        // delete list range :  [     ]
```

Comment:
logic need change maybe

Reply:
> logic need change maybe

I adjusted the logic according based on our discussion.